### DISCUSSION TITLE
Cannot "import colour" with Python 2.7.12 64-bit.

### DISCUSSION THREAD

**Discussion by Kunkka1988** [colour-science/colour#703] [Q&A] [answered]
Hi KelSolaar,

I got a trouble with colour-science.
The colour-0.3.10.zip had been downloaded and installed, but when I tried to "import colour" on python 2.7.12 64-bit that will be failed. 
If I use the 32-bit python then it will be fine, but I need the 64-bit python to work.

Here are the hints:
...
>>> import colour
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\lib\site-packages\colour_science-0.3.10-py2.7.egg\colour\__init__.py", line 41, in <module>
    from .algebra import *  # noqa
  File "C:\Python27\lib\site-packages\colour_science-0.3.10-py2.7.egg\colour\algebra\__init__.py", line 12, in <module>
    from .interpolation import (LinearInterpolator, SpragueInterpolator,
  File "C:\Python27\lib\site-packages\colour_science-0.3.10-py2.7.egg\colour\algebra\interpolation.py", line 21, in <module>
    import scipy.interpolate
  File "C:\Python27\lib\site-packages\scipy\__init__.py", line 116, in <module>
    from scipy._lib._ccallback import LowLevelCallable
  File "C:\Python27\lib\site-packages\scipy\_lib\_ccallback.py", line 1, in <module>
    from . import _ccallback_c
ImportError: DLL load failed: %1 is not a valid Win32 application.
...

Could you have a look at this issue if free?
Thanks.

BR,

Kunkka

**Comment by thomasmansencal** [comment, marked as answer]
@Kunkka1988 : It seems like you have an installation issue with Scipy and this is not a `Colour` issue but a broad one in regard to your Python installation:

``` python
File "C:\Python27\lib\site-packages\scipy_lib_ccallback.py", line 1, in 
from . import _ccallback_c
ImportError: DLL load failed: %1 is not a valid Win32 application.
```

Scipy has compiled libraries and it seems like you are trying to load one that has architecture mismatch with the Python interpreter you are intended to use it with.

Please double check that both your Numpy and Scipy versions are compliant with your interpreter. You can directly import a Numpy and Scipy module at Python prompt to check they are working:

``` python
import numpy
import scipy.interpolate
```

More informations are available in that thread: https://stackoverflow.com/questions/19019720/importerror-dll-load-failed-1-is-not-a-valid-win32-application-but-the-dlls

**Comment by Kunkka1988** [comment]
Hi KelSolaar,

It works well after reinstall scipy-0.19.1-cp27-cp27m-win_amd64.whl from http://www.lfd.uci.edu/~gohlke/pythonlibs/#numpy.
Thank you so much:)

**Comment by thomasmansencal** [comment]
You are welcome @Kunkka1988! :)

**Comment by dtbulmerJRs** [comment]
I solved it by copying the module colour.py from the python2 directory to the python3 directory? took me weeks of trying things out to fix it

'''
sudo cp /usr/local/lib/python2.7/dist-packages/colour.py /usr/local/lib/python3.7/dist-packages/colour.py
'''

**Comment by thomasmansencal** [comment]
Hi @dtbulmerJRs,

You should avoid installing anything in your system Python environment, prefer a Virtual Environment ideally.

Cheers,

Thomas
